<launch>
    <!-- EXPERIMENT NAME -->
    <arg name="id" />


    <!-- SYSTEM MODEL -->
    <!-- vehicle parameters -->
    <param name="L_a" type="double" value="0.18" />
    <param name="L_b" type="double" value="0.14" />
    <param name="r_tire" type="double" value="0.05" />
    <param name="psio_gps" type="double" value="80.0" />
    <param name="sim_mode" type="bool" value="false" />
	
	<!-- ARDUINO -->
    <!--  * encoders and ultrasound sensors -->
    <node pkg="rosserial_python" type="serial_node.py" name="arduino_node" >
        <param name="port" value="/dev/ttyUSB0" />
    </node>

	<!-- GPS -->
	<node pkg="nmea_navsat_driver" type="nmea_serial_driver" name="gps">
        <param name="port" value="/dev/ttyACM1" />
        <param name="baud_rate" value="115200" />
    </node>

    <!-- IMU -->
    <node pkg="myahrs_driver" type="myahrs_driver" name="myahrs_driver">
        <param name="port" value="/dev/ttyACM0" />
        <param name="baud_rate" value="115200" />
    </node>

    <!-- OBSERVER-->
    <node pkg="barc" type="state_estimation_KinBkMdl.py" name="state_estimation">
        <!-- v_x estimation sample time-->
        <param name="dt_v_enc" type="double" value="0.2" />

        <!-- ekf properties -->
        <param name="q_std" type="double" value="0.1" />
        <param name="r_std" type="double" value="0.1" />

        <!-- measurement model type -->
        <param name="est_mode" type="int" value="1" />
    </node>

    <!-- LOW LEVEL CONTROLLER -->
    <node pkg="barc" type="controller_low_level.py" name="low_level_controller" />

    <!-- MPC CONTROLLER -->
    <node pkg="barc" type="controller_MPC_traj.jl" name="MPC" output="screen"/>

    <!-- RECORD -->
    <node pkg="barc" type="record_experiment.py" name="record_experiment" output="screen">
        <param name="experiment_name" value="$(arg id)" />
        <param name="camera_on" value="false" />
    </node>

</launch>
